### PR TITLE
Feature/FindNetCDF fix for sperate c and fortran installation

### PR DESCRIPTION
**Pull request by xinzhang8noaa**
The original cmake/FindNetCDF can not locate NetCDF C/Fortran properly when NetCDF C and Fortran are installed in different prefix. For example : Spack installs NetCDF C/ Fortran in different prefix. I found this problem when using Spack installed environment on NOAA Jet.

This version will determine the location of C/Fortran correctly in every know situation.

Now, only NetCDF C and Fortran are considered.

### CODE REVIEW

**Comment by markjolah** [comment]
Just talked with @xinzhang8noaa.  I'm going to merge this functionality with #24 to make a single PR for NetCDF.

File: cmake/FindNetCDF.cmake [hunk 1]
```diff
@@ -79,7 +83,7 @@ foreach( _comp ${${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS} )
   endif()
 endforeach()
 if( NOT _search_components )
-  set( _search_components C )
+  set( _search_components C Fortran )
```
Comment (aerorahul) [inline cmake/FindNetCDF.cmake]: Only search for `Fortran` if requested.  By default, `C` is the only component to look for.

**Comment by xinzhang8noaa** [comment]
Trying to extened to C++

**Comment by mmiesch** [comment]
@xinzhang8noaa - should we close this as we discussed at the meeting?  Has this functionality been incorporated into https://github.com/JCSDA/ecbuild/pull/24?

**Comment by xinzhang8noaa** [comment]
Sure, Thanks!

**Pull request closed without merging**